runtime/interpreter/interpreter.go

```diff
@@ -1614,181 +1616,144 @@ func (interpreter *Interpreter) VisitIdentifierExpression(expression *ast.Identi
 	return Done{Result: variable.Value}
 }
 
-// valueTuple
-
-type valueTuple struct {
-	left, right Value
-}
-
-// visitBinaryOperation interprets the left-hand side and the right-hand side and returns
-// the result in a valueTuple
-func (interpreter *Interpreter) visitBinaryOperation(expr *ast.BinaryExpression) Trampoline {
-	// interpret the left-hand side
-	return expr.Left.Accept(interpreter).(Trampoline).
-		FlatMap(func(left interface{}) Trampoline {
-			// after interpreting the left-hand side,
-			// interpret the right-hand side
-			return expr.Right.Accept(interpreter).(Trampoline).
-				FlatMap(func(right interface{}) Trampoline {
-					tuple := valueTuple{
-						left.(Value),
-						right.(Value),
-					}
-					return Done{Result: tuple}
-				})
-		})
-}
-
-func (interpreter *Interpreter) visitNumberBinaryOperation(
-	expression *ast.BinaryExpression,
-	f func(left, right NumberValue) Value,
-) ast.Repr {
-	return interpreter.visitBinaryOperation(expression).
-		Map(func(result interface{}) interface{} {
-			tuple := result.(valueTuple)
-			left := tuple.left.(NumberValue)
-			right := tuple.right.(NumberValue)
-			return f(left, right)
-		})
+func (interpreter *Interpreter) visitBinaryOperation(
+	expr *ast.BinaryExpression,
+	f func(left, right Value) Value,
+) Trampoline {
+	left := interpreter.runAllStatements(expr.Left.Accept(interpreter).(Trampoline)).(Value)
+	right := interpreter.runAllStatements(expr.Right.Accept(interpreter).(Trampoline)).(Value)
```

Comment:
Since we will be writing this line a lot going forward, maybe its worth extracting this logic to a method. 
```go
func (interpreter *Interpreter) evaluate(element ast.Element) Value {
    return interpreter.runAllStatements(element.Accept(interpreter).(Trampoline)).(Value)
}
```

Then these lines simplify to:
```go
left := interpreter.evaluate(expr.Left)
right := interpreter.evaluate(expr.Right)
```

Reply:
Yep 👍  I refactored that in the next PR: https://github.com/onflow/cadence/pull/681/files#diff-a36f730b18ca31316909bd74125003c1fbe785a13f926b4a0631e7c0cc6a51e4R1552